sympy/solvers/solvers.py

```diff
@@ -1008,6 +1014,9 @@ def _sympified_list(w):
                 f[i: i + 1] = [fr, fi]
 
     # real/imag handling -----------------------------
+    if bool_check:
+        return []
```

Comment:
Can we not just do `if any(isinstance(fi, bool) for fi in f)` somewhere here?

Reply:
Or just `if False in f`

Reply:
Yes also it would be more cleaner to write it this way. But `if False in f` would yield wrong result as `0` is also considered as `False` in python.